perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -244,6 +269,19 @@ def CleanupBucket(storage_schema):
     objects_to_cleanup = _ListObjects(storage_schema, FLAGS.bucket, prefix=None)
 
 
+def GenerateWritePayload(size):
+  payload_bytes = bytearray(size)
+  # don't use a for...range(100M), range(100M) will create a list of 100mm items
+  # with each item being the default object size of 30 bytes or so, it will lead
+  # to out of memory error.
+  for i in xrange(size):
+    payload_bytes[i] = ord(random.choice(string.letters))
+
+  payload_string = payload_bytes.decode('ascii')
+
+  return payload_bytes, payload_string
```

Comment:
I did it because I was keeping the interface exactly the same as before. I strongly suspect the benchmark currently uses 2x the memory necessary. Looking at the code, the reason is that the Azure backend uses the bytes variable, and the boto backend uses the string variable.

I'm preparing a separate PR in which I upgrade our S3 support to boto3 (it's preparation for supporting multiple storage classes). As part of that PR, I plan to move different backends into different classes to use Python's polymorphism instead of `if` statements and `_useBotoApi()` (there will be 3 backends then, and I think that will be much more clean). I think that is the most natural place to fix this issue.